R/finalize_plot.R

```diff
@@ -135,21 +140,25 @@ finalize_plot <- function(plot = NULL,
   # check args with default vectors
   caption_valign <- match.arg(caption_valign)
 
-  # check mode argument and validate filename
+  # check mode argument
   savetypes_raster <- c("png","tiff","jpeg","bmp")
   savetypes_vector <- c("svg","ps","pdf")
   savetypes_print <- c("plot", "window")
 
   mode <- match.arg(arg = mode,
-                    choices = c(savetypes_raster,
+                    choices = c(savetypes_print,
                                 savetypes_vector,
-                                savetypes_print,
-                                "object"),
+                                savetypes_raster),
                     several.ok = TRUE)
 
+  # remove any `window` mode specified if OS is not Windows
+  if (.Platform$OS.type != "windows"){
+    mode <- mode[mode != "window"]
+  }
```

Comment:
I'd like to request different behavior here, if it's not too much work. It's not great practice to implement arguments in a way that code will work on one system and then cause a hard crash if run on a different one. Here's my suggestion: allow `mode="window"` as a valid parameter on non-Windows, but switch the mode to `"plot"` at runtime and produce a warning like "mode='window' is not supported on non-Windows systems. Switching to mode='plot' instead."

Reply:
Does this cause a crash? The goal was to simply remove "window" from the list of modes and continue with the function otherwise. This would mean no modes are listed, I guess, if  "window" is the only argument passed. Maybe that's where the crash comes from? If so, the easiest solution may be to move this conditional to above `match.arg` and add the warning. That way, `mode = NULL` will be converted to `mode = "plot"` by `match.arg` if it is not already in the list.

Happy to tweak... can you clarify what behavior you're currently experiencing though?

Reply:
I hadn't tested and was assuming it would fail if I ran `finalize_plot(mode="window")` on macOS. I just tested and it didn't crash *per se*, but no plot was generated and no message was provided. Here's my revised request:

- If `mode` includes `"window"` but not `"plot"`, replace `"window"` with `"plot"` so that a plot is generated, and provide a warning.
- If `mode` includes `"window"` *and* `"plot"`, remove `"window"` from the list (current behavior). Also provide a (different) warning in this case.

Reply:
Let me know if the commits I just pushed work well on your system

Reply:
Works as advertised